test_communication/test/test_timeout_subscriber.cpp

```diff
@@ -0,0 +1,74 @@
+// Copyright 2015 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <chrono>
+#include <iostream>
+#include <stdexcept>
+#include <string>
+
+#include <rclcpp/rclcpp.hpp>
+
+#include "message_fixtures.hpp"
+
+int retcode;
```

Comment:
Can I just throw a runtime error in the callback instead, since it's not expecting to receive a message?